packages/shared/src/components/modals/bookmark/BookmarkFolderModal.tsx

```diff
@@ -0,0 +1,221 @@
+            event_name: LogEvent.UpgradeSubscription,
+            target_id: TargetId.BookmarkFolder,
+          });
+        }}
+      >
+        <Typography
+          tag={TypographyTag.Span}
+          type={TypographyType.Callout}
+          color={TypographyColor.Plus}
+          className="underline"
+        >
+          upgrade to Plus
+        </Typography>
+      </Link>
+    </Typography>
+  );
+};
+
+const FormActions = ({
+  valid,
+  folderCount,
+  isPlus,
+  isMobile,
+}: {
+  valid: boolean;
+  folderCount: number;
+  isPlus: boolean;
+  isMobile: boolean;
+}) => {
+  const { closeModal } = useLazyModal();
+
+  return (
+    <div
+      className={classNames(
+        'flex w-full justify-between px-4 pt-4 tablet:px-0 tablet:pt-0',
+        isMobile && 'border-b border-border-subtlest-tertiary pb-4',
+      )}
+    >
+      {isMobile && (
+        <Button
+          type="button"
+          onClick={closeModal}
+          variant={ButtonVariant.Tertiary}
+        >
+          Cancel
+        </Button>
+      )}
+      <Button
+        type="submit"
+        className="tablet:flex-grow"
+        disabled={!valid}
+        variant={ButtonVariant.Primary}
+      >
+        {!isPlus && folderCount > 0 ? (
+          <>
+            <DevPlusIcon /> Upgrade to plus
+          </>
+        ) : (
+          'Create folder'
+        )}
+      </Button>
+    </div>
+  );
+};
+
+const BookmarkFolderModal = ({
+  folder,
+  onSubmit,
+  folderCount = 0,
+  ...rest
+}: BookmarkFolderModalProps): ReactElement => {
+  const { isPlus } = usePlusSubscription();
+  const [valid, setValid] = useState(false);
+  const isMobile = useViewSize(ViewSize.MobileL);
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // TODO: Dummy component for now. Implement request later.
+    // const formJson = formToJson(e.currentTarget);
+  };
+
+  return (
+    <Modal kind={Modal.Kind.FlexibleCenter} size={Modal.Size.Small} {...rest}>
+      <form onSubmit={handleSubmit}>
+        {isMobile && (
+          <FormActions
+            valid={valid}
+            folderCount={folderCount}
+            isPlus={isPlus}
+            isMobile={isMobile}
+          />
+        )}
+        <ModalHeader
+          showCloseButton={!isMobile}
+          className={classNames('gap-2', isMobile && '!border-b-0 pb-0')}
+        >
+          <ModalHeader.Title>New Folder</ModalHeader.Title>
+          <Typography
+            tag={TypographyTag.Span}
+            type={TypographyType.Caption1}
+            className="flex items-center rounded-4 bg-action-plus-float px-1 font-bold"
+            color={TypographyColor.Plus}
+          >
+            <DevPlusIcon size={IconSize.Size16} />
+            Plus
+          </Typography>
+        </ModalHeader>
+        <Modal.Body className="flex flex-col gap-5 tablet:gap-4">
+          {!isPlus && <PlusCTA folderCount={folderCount} />}
+          <TextField
+            maxLength={50}
```

Comment:
This is a hard limit, if so it needs to be on API as well.

Reply:
@ilasw This is from the design. Are we going to actually support a character limit, or should I remove this?

![image](https://github.com/user-attachments/assets/57744548-2ded-4b05-b3e3-59ca5123081f)

Reply:
Just note to add backlog item for API to limit :)